http-clients/aws-crt-client/src/it/java/software/amazon/awssdk/http/crt/AwsCrtClientCallingPatternIntegrationTest.java

```diff
@@ -109,38 +108,36 @@ private boolean testWithNewClient(int eventLoopSize, int numberOfRequests) {
         }
     }
 
-    @DataPoints("EventLoop")
-    public static int[] eventLoopValues(){
-        return new int[]{1, 4};
-    }
-
-    @DataPoints("ConnectionPool")
-    public static int[] connectionsValues(){
-        /* Don't use 1 connection Pool of size 1, otherwise test takes too long */
-        return new int[]{10, 100};
-    }
-
-    @DataPoints("NumRequests")
-    public static int[] requestValues(){
-        return new int[]{1, 25, 250};
-    }
-
-    @DataPoints("ParallelClients")
-    public static int[] parallelClientValues(){
-        return new int[]{1, 2, 8};
-    }
-
-    @DataPoints("SharedClient")
-    public static boolean[] sharedClientValue(){
-        return new boolean[]{true, false};
+    static Integer[] eventLoopValues = {1, 4};
+    /* Don't use 1 connection Pool of size 1, otherwise test takes too long */
+    static Integer[] connectionsValues = {10, 100};
+    static Integer[] requestValues = {1, 25, 250};
+    static Integer[] parallelClientValues = {1, 2, 8};
+    static Boolean[] sharedClientValue = {true, false};
+
+    private static Stream<Arguments> providePossibleCallPatternsWithDifferentConfigs() {
```

Comment:
Nice way of providing the combinations. How about
~~~
    private static Stream<Arguments> permutationsOfCrtCallParameters() {
        return Arrays.stream(eventLoopValues).flatMap(
            eventLoops -> Arrays.stream(connectionsValues).flatMap(
                connections -> Arrays.stream(requestValues).flatMap(
                    requests -> Arrays.stream(parallelClientValues).flatMap(
                        parallelClients -> Arrays.stream(sharedClientValue).map(
                            sharedClients -> Arguments.of(eventLoops, connections, requests, parallelClients, sharedClients))))));
    }
~~~
Including the parameter names instead of `s1` etc makes it less abstract, and also collecting the parenthesis at the end is easier to read